src/D2L.CodeStyle.TestAnalyzers/IgnoreAttribute/IgnoreAttributeAnalyzer.cs

```diff
@@ -41,6 +42,15 @@ internal sealed class IgnoreAttributeAnalyzer : DiagnosticAnalyzer {
 
 			foreach( var attribute in root.Attributes ) {
 				if( attribute.Name.ToString().Equals( "Ignore" ) ) {
+					var parentClassDeclaration = root.AncestorsAndSelf().OfType<ClassDeclarationSyntax>().ToImmutableArray().Single();

```

Comment:
I think a better approach might be to check what the ignore attribute is on, and make sure it's a method. 
And to check to make sure that the attribute class's namespace is the nunit one.